tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -0,0 +1,925 @@
+            {
+                x ^= data[idx];
+            }
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Indexer in-loop bounds check elimination")]
+        public static void ReadOnlyIndexer2(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                ReadOnlySpan<byte> s = new ReadOnlySpan<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = TestReadOnlyIndexer2(s);
+                }
+                return result;
+            },
+            "ReadOnlyIndexer2({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestReadOnlyIndexer2(ReadOnlySpan<byte> data)
+        {
+            byte x = 0;
+
+            for (var idx = 0; idx < data.Length; idx++)
+            {
+                x ^= data[idx];
+            }
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Indexer in-loop bounds check elimination w/ writes")]
+        public static void WriteViaIndexer1(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = WriteViaIndexer1(s);
+                }
+                return result;
+            },
+            "WriteViaIndexer1({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte WriteViaIndexer1(Span<byte> data)
+        {
+            byte q = data[0];
+
+            for (var idx = 1; idx < data.Length; idx++)
+            {
+                data[0] ^= data[idx];
+            }
+
+            byte x = data[0];
+            data[0] = q;
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Indexer in-loop bounds check elimination w/ writes")]
+        public static void WriteViaIndexer2(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = WriteViaIndexer2(s);
+                }
+                return result;
+            },
+            "WriteViaIndexer2({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte WriteViaIndexer2(Span<byte> data)
```

Comment:
Calling this `TestWriteViaIndexer2` would be more consistent with the rest of the naming.  (the others besides WriteViaIndexer1/WriteViaIndexer2 look fine)